src/project/internal/projectactionscontroller.cpp

```diff
@@ -972,7 +972,15 @@ void ProjectActionsController::prependToRecentScoreList(const io::path_t& filePa
     }
 
     recentScorePaths.insert(recentScorePaths.begin(), filePath);
+
+    static constexpr size_t MAX_RECENT_FILES = 100;
+
+    if (recentScorePaths.size() > MAX_RECENT_FILES) {
+        recentScorePaths.resize(MAX_RECENT_FILES);
+    }
+
     configuration()->setRecentProjectPaths(recentScorePaths);
```

Comment:
that's OK for configuration to know about all the possible recent files. The real problem is in model/view. Move this limitation into "RecentProjectsModel"

Reply:
> that's OK for configuration to know about all the possible recent files

Really? Every time that we want to add (or "bump") a recent file, we need to search through the whole list, to find if the file is already in the list, and if yes, erase it from there. If the list grows longer and longer, this will take more and more time. Of course I understand that this amount of time will likely still be negligible for most users, but it just seems wasteful. Especially if we never actually show all recent files to the user, so it has completely zero benefit to store them.

If you don't like to have this logic here, we can use an alternative solution; in `ProjectConfiguration::setRecentProjectPaths`, where the list is actually saved, we iterate over the list of paths to convert them to JSON. Instead of iterating over all of them, we could iterate over the first 100 only. That seems not much better to me, but it's just an alternative.

Reply:
Again, that's a responsibility of model/view. The original data should remain consistent.

Usually, you'd expect to have "technical" limitation which user would never reach, but you still want to have "some" upper bound (let's say 1000) and "visual" limitation of model/view which saves some performance for you.

The reason why I'm asking you to differentiate this is pretty simple - let's say, tomorrow we'll decide to show 150 items instead of 100, since the "original" data would remain consistent you'll not miss anything

> Every time that we want to add (or "bump") a recent file, we need to search through the whole list, to find if the file is already in the list

BTW If that's a problem you were going to resolve by that limit, I'd suggest to change a container on unordered_set/map and do the necessary sorting on the model's side